pkg/api/cortex-ruler.go

```diff
@@ -239,11 +298,18 @@ type GettableExtendedRuleNode struct {
 	GrafanaManagedAlert *GettableGrafanaRule `yaml:"grafana_alert,omitempty" json:"grafana_alert,omitempty"`
 }
 
-func (n *GettableExtendedRuleNode) Type() RuleType {
-	if n.ApiRuleNode != nil {
-		return LoTexManagedRule
+func (n *GettableExtendedRuleNode) Type() (RuleType, error) {
+	if n.ApiRuleNode == nil && n.GrafanaManagedAlert == nil {
+		return 0, fmt.Errorf("cannot have empty rule")
 	}
-	return GrafanaManagedRule
+
+	if n.GrafanaManagedAlert != nil {
```

Comment:
I'm not sure this validation belong in `Type`, is there any particular reason for it? Otherwise, I'd be inclined for following the pattern throughout this file and creating a function called `validate` that we can use as we unmarshal.

Reply:
The only reason is that Type() and validate() will end up very similar but I will change it.